### PR TITLE
Add `woocommerce_order_note_deleted` hook for a successful order note…

### PR DESCRIPTION
**Pull request by evokelektrique**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

As it was written in the README.md, we should have the least possible changes in the legacy code section, and I made a very small change in the function of deleting order notes, and this change can be very useful, such as synchronizing the store with external systems. And in the pull request, I have tried to write the least possible changes of a hook for a successful order note removal.

### CODE REVIEW

File: plugins/woocommerce/includes/wc-order-functions.php [hunk 1]
```diff
@@ -1148,5 +1148,18 @@ function wc_create_order_note( $order_id, $note, $is_customer_note = false, $add
...
+
+	if($delete) {
+		/**
+		 * Action hook fired after an order note is deleted.
+		 *
+		 * @param int      $note_id	Order note ID.
+		 *
+		 * @since 7.7.0
+		 */
+		do_action( 'woocommerce_order_note_deleted', $note_id);
```
Comment (vedanshujain) [inline plugins/woocommerce/includes/wc-order-functions.php:1161]: If we are executing an action, let's fetch the `$note` object, and pass it as a parameter. Just the note_id which is deleted is probably not very useful, especially in systems which already don't have another copy of the note identified by the ID.

**Review by vedanshujain — COMMENTED** [review]
Just curious, why not use the `deleted_comment` action here instead?

File: plugins/woocommerce/includes/wc-order-functions.php (hunk 1 above)
Comment (evokelektrique) [reply plugins/woocommerce/includes/wc-order-functions.php:1161]: I agree, but could you help me with that, how do i fetch the `$note` object? and what about `deleted_comment`? of course order notes are WordPress comments but I just need it to be more specific.

Comment (rrennick) [reply plugins/woocommerce/includes/wc-order-functions.php:1161]: @evokelektrique I apologize for the delay in follow up. Use `$note = wc_get_order_note( $note_id );` before line 1151.

Comment (evokelektrique) [reply plugins/woocommerce/includes/wc-order-functions.php:1161]: I'll get into it

File: plugins/woocommerce/includes/wc-order-functions.php [hunk 2]
```diff
@@ -1148,5 +1148,18 @@ function wc_create_order_note( $order_id, $note, $is_customer_note = false, $add
  * @return bool         True on success, false on failure.
  */
 function wc_delete_order_note( $note_id ) {
-	return wp_delete_comment( $note_id, true );
+	$delete = wp_delete_comment( $note_id, true );
+
+	if($delete) {
```
Comment (jorgeatorres) [inline plugins/woocommerce/includes/wc-order-functions.php]: Spacing here is incorrect. 
```suggestion
	if ( $delete ) {
```

File: plugins/woocommerce/includes/wc-order-functions.php [hunk 3]
```diff
@@ -1148,5 +1148,18 @@ function wc_create_order_note( $order_id, $note, $is_customer_note = false, $add
...
+		/**
+		 * Action hook fired after an order note is deleted.
+		 *
+		 * @param int      $note_id	Order note ID.
+		 *
+		 * @since 7.7.0
```
Comment (jorgeatorres) [inline plugins/woocommerce/includes/wc-order-functions.php:1159]: This version here should be 8.5.0 now.

**Review by jorgeatorres — CHANGES_REQUESTED** [review]
Hi @evokelektrique!

Thanks again for your contribution. As https://github.com/woocommerce/woocommerce/pull/38498#discussion_r1212612257 we would just need a small change on your side, as well as some tiny adjustments that I've highlighted as part of the review.

Are you still interested in making those changes yourself? If you're currently short on time and would like us to get this across the finish line, please say so. You'd still retain attribution for the original commits. FWIW, this does seem like a good addition to our codebase.

Looking forward to hearing back from you!

File: plugins/woocommerce/includes/wc-order-functions.php (hunk 3 above)
Comment (evokelektrique) [reply plugins/woocommerce/includes/wc-order-functions.php:1159]: Hi, sure I'll change it

**Comment by jorgeatorres** [comment]
Superseded by #47916

**Pull request closed without merging**